front/app/containers/Admin/reporting/components/ReportBuilder/Widgets/ChartWidgets/MethodsUsedWidget/ChartWidgetSettings.tsx

```diff
@@ -1,27 +1,12 @@
 import React from 'react';
 
-import { useNode } from '@craftjs/core';
-
 import { DateRangeInput } from '../_shared/ChartWidgetSettings';
 import { ComparisonToggle } from '../_shared/StatisticToggles';
 
 const ChartWidgetSettings = () => {
-  const { setProp } = useNode();
-
   return (
     <>
-      <DateRangeInput
-        onChangeDateRange={({ startDate, endDate }) => {
-          if (!startDate || !endDate) {
-            // Make sure that we always reset compared date range
-            // if the main date range is not fully set
-            setProp((props) => {
-              props.compareStartAt = undefined;
-              props.compareEndAt = undefined;
-            });
-          }
-        }}
-      />
+      <DateRangeInput resetComparePeriod />
```

Comment:
Maybe, it would be better to keep `DateRangeInput` and `TimeSeriesWidgetSettings` decoupled from the comparison settings. Can we still use them in contexts where there is no comparison involved?

Reply:
That's how I set it up originally, but it just meant I had to copy and paste the same logic many times over. I think we need this particular comparison logic in enough places to justify adding a prop to easily add it like this.

The component is also just a wrapper around our existing `DateRangePicker`, so if we want to use this outside the report builder we can just use that one instead